data/publiccloud/terraform/gce.tf

```diff
@@ -78,6 +78,16 @@ variable "enable_confidential_vm" {
 	default=false
 }
 
+variable "gpu" {
```

Comment:
If you will just tweak it to :
```
guest_accelerator {
      type = "nvidia-tesla-t4"
      count = var.gpu ? 1 : 0
}
``` 

and 

```
variable "gpu" {
 default=false
}
```
it will gives you same result but will be much easy to read and understand what is going on

Reply:
you don't need to reset type in case count is 0

Reply:
> If you will just tweak it to :
> 
> ```
> guest_accelerator {
>       type = "nvidia-tesla-t4"
>       count = var.gpu ? 1 : 0
> }
> ```
> 
it didnt work. As is, it is a much better approach IMO.

> and
> 
> ```
> variable "gpu" {
>  default=false
> }
> ```
> 
> it will gives you same result but will be much easy to read and understand what is going on

Reply:
looks like we have two different discussions here : 
1. It didn't work - can you point me to failed job where you trying to use this approach so I can see the problem ?
2. Can you elaborate what is "better" means to you here ? You don't win anything if type would be null or have some value when you not using this section at all

Reply:
also please provide a VR of your code when gpu is not set , I have doubts about 
```
        for_each = var.gpu[*]
            content {
                type  = guest_accelerator.value.type
                count = guest_accelerator.value.count
            }
```

when gpu is null that it won't throw exception ...

Reply:
> looks like we have two different discussions here :
> 
> 1. It didn't work - can you point me to failed job where you trying to use this approach so I can see the problem ?

http://aquarius.suse.cz/tests/13543#step/prepare_instance/71

> 2. Can you elaborate what is "better" means to you here ? You don't win anything if type would be null or have some value when you not using this section at all

IIUC with `null` as a default, terraform ignores the resource. if resource's value is not `null` then it uses it. In the other had a boolean `false` needs that conditional logic.

Reply:
> also please provide a VR of your code when gpu is not set , I have doubts about
> 
> ```
>         for_each = var.gpu[*]
>             content {
>                 type  = guest_accelerator.value.type
>                 count = guest_accelerator.value.count
>             }
> ```
> 
> when gpu is null that it won't throw exception ...

http://aquarius.suse.cz/tests/13490#step/prepare_instance/56

Reply:
> > looks like we have two different discussions here :
> > 
> > 1. It didn't work - can you point me to failed job where you trying to use this approach so I can see the problem ?
> 
> http://aquarius.suse.cz/tests/13543#step/prepare_instance/71

`-var 'gpu={"count":1,"type":"nvidia-tesla-t4"}'`  it is not what you suppose to pass when  

```
guest_accelerator {
      type = "nvidia-tesla-t4"
      count = var.gpu ? 1 : 0
}
```
you need to pass true/false than everything will work as expected !  So this run can not prove that "it didn't work" because you defined boolean variable and trying to pass dict there 

> IIUC with `null` as a default, terraform ignores the resource. if resource's value is not `null` then it uses it. In the other had a boolean `false` needs that conditional logic.
```
guest_accelerator {
      type = "nvidia-tesla-t4"
      count = 0
}
```

this WILL be ignored no matter that type has value

Reply:
*ready for review*

Reply:
for the record, now the disadvantage is that the type is hardcoded. My previous approach it was given the agility to keep the terraform file the same and pass the variable as you like.